daisy/step_attach_disks.go

```diff
@@ -85,8 +85,12 @@ func (a *AttachDisks) validate(ctx context.Context, s *Step) (errs DError) {
 		ad.zone = disk["zone"]
 
 		// Register disk attachments.
-		errs = addErrs(errs, s.w.instances.w.disks.regAttach(ad.Source, ad.Instance, ad.Mode, s))
-
+		// If device name is not given explicitly, its device name will be the same as disk name.
+		deviceName := ad.Source
+		if ad.DeviceName != "" {
+			deviceName = ad.DeviceName
+		}
```

Comment:
ad.Source is disk name? Shouldn't the order of initialization be reversed here then? First set ad.DeviceName, if it's empty, set ad.Source? Or this is a different use case?

This is the third place where this device/disk default logic is implemented, so another reason to move it to regAttach, if it's possible to be generalized.

Reply:
done